py-polars/polars/utils/udfs.py

```diff
@@ -260,14 +262,16 @@ def to_expression(self, col: str, as_repr: bool = True) -> str | None:
             },
             logical_instructions,
         )
-        polars_expr = " ".join(expr for _offset, expr in expression_blocks)
+        polars_expr = " ".join(expr for _offset, expr in expression_strings)
 
         # note: if no 'pl.col' in the expression, it likely represents a compound
         # constant value (e.g. `lambda x: CONST + 123`), so we don't want to warn
         if "pl.col(" not in polars_expr:
             return None
-
-        return polars_expr if as_repr else eval(polars_expr, globals())
+        elif self._apply_target == "series":
+            return polars_expr.replace(f'pl.col("{col}")', "s")
```

Comment:
what if there's a global variable called `s`? might this conflict with that?

this is the case I'm worried about:
```python
In [5]: s = 3

In [6]: df = pl.DataFrame({'a': [1,2,3]})

In [7]: df['a'].apply(lambda x: x+s)
```

Reply:
Well played, lol... Yes, that would be an unlikely but real issue! I'll see about mitigating it, hmm 🤔